File: state/c-sharp/VendingMachineState.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace State
{
    // Davranışların tanımlandığı soyut sınıftır.
    // UML diyagramındaki State yapısına denk gelmektedir.
    public abstract class VendingMachineState
    {
        public abstract void HandleState(VendingMachine context);
    }
}

File: state/typescript/vending-machine-state.ts
import {VendingMachine} from "./vending-machine";

// Davranışların tanımlandığı soyut sınıftır.
// UML diyagramındaki State yapısına denk gelmektedir.
export abstract class VendingMachineState {
  public abstract handleState(context: VendingMachine): void;
}
